### ISSUE TITLE
Zoom to fit resulting in client side crash for some large pipelines

### ISSUE THREAD

**Issue by chrisj-ibm** [elyra-ai/canvas#1047]
In Watson Studio pipelines, we're seeing a client side crash when using the `zoom to fit` function on some very large pipelines:
![zoom-to-fit-error](https://user-images.githubusercontent.com/105053616/167127573-0a5b501a-1163-4ff5-b63c-cc3bbb665908.gif)

The crash does not happen every time zoom to fit is invoked, only some of the time, but we haven't been able to isolate why.  I thought it might be related to the validation / save flow, but we do not have any issue resizing while validation is running on smaller pipelines.  

This was discussed previously in IBM slack here: https://ibm-analytics.slack.com/archives/C598XEX8S/p1650897161402039.

The associated IBM internal issue is: https://github.ibm.com/AILifecycle/tracker/issues/3340.

When the crash occurs, the console errors look like this: 
![image](https://user-images.githubusercontent.com/105053616/167124774-4af8ced7-bf18-4e78-9463-2c06fef511dd.png)

The associated pipeline json has been provided in the internal issue here: https://github.ibm.com/AILifecycle/tracker/issues/3340#issuecomment-43517349.  

Please advise if you need any additional information from our side.  Thanks.

**Comment by tomlyn** [comment]
@chrisj-ibm 
I believe this is caused by the zoom to fit calculating a zoom scale amount which is smaller than that allowed by the schema. This can happen with a large number of nodes so the nodes occupy a lot of coordinate space. This smaller-than-allowed scale value is then saved in the pipeline flow document -- I'm assuming you are running with the config property `enableSaveZoom` set to `"Pipelineflow"`.  When the pipeline flow is loaded back into common-canvas the schema validation throws an error because it sees the scale amount is too small. 

I have fixed it by making the pipeline flow, that is returned by common-canvas, always conform to the schema. When do you need the fix. I can do a patch release next week if necessary.

**Comment by chrisj-ibm** [comment]
@tomlyn A patch release next week would be amazing if you can contain it.  We would like to get this fix into our 4.5 release, with a target date of next Friday for our release candidate.  If not containable, we can document this particular issue as a known issue.  Thanks so much for the prompt support, as always.

**Comment by tomlyn** [comment]
@chrisj-ibm 
OK I'll see what I can do.
Are you running with `schemaValiation` config property set to `true` when running in production? That feature can slow down loading of pipeline flows considerably when there are a lot of nodes.

**Comment by chrisj-ibm** [comment]
> Are you running with schemaValiation config property set to true when running in production? That feature can slow down loading of pipeline flows considerably when there are a lot of nodes.

Yes, appears so.  I will discuss this further with Rafal to see if we want to revise our approach here.

**Comment by tomlyn** [comment]
@chrisj-ibm 
Hi Chris -- I realized my fix to the code might not work 100% of the time so I decided to go in a different direction to resolve this issue. So now I have changed the pipeline schema file to remove the zoom scale restrictions. This means that any zoom scale amount in the pipeline flow document can be read in, with schema validation switched on, without a problem.

To pick up this change you don't need a patch release, you just need to do an `npm install` to refresh your node_modules which should pick up the latest @elyra/pipeline-schemas (v3.0.57) which will contain the change.
